virtool/groups/db.py

```diff
@@ -99,53 +99,50 @@ async def fetch_group_users(db, group_id: str) -> List[UserNested]:
     ]
 
 
-def lookup_minimal_groups(
-    local_field: str = "groups", set_as: str = "groups"
-) -> list[dict]:
+def lookup_group_minimal_by_id(
+    local_field: str = "group", set_as: str = "group"
+) -> List[dict]:
     """
-    Create a mongoDB aggregation pipeline step to look up minimal groups.
+    Return a list of aggregation pipeline stages to lookup a group by its id and return
+    only the ``_id`` and ``name`` fields.
 
-    :param local_field: groups field to look up
-    :param set_as: desired name of the returned record
-    :return: mongoDB aggregation steps for use in an aggregation pipeline
     """
     return [
         {
             "$lookup": {
                 "from": "groups",
-                "let": {"group_ids": f"${local_field}"},
+                "let": {"group_id": f"${local_field}"},
                 "pipeline": [
-                    {"$match": {"$expr": {"$in": ["$_id", "$$group_ids"]}}},
-                    {"$sort": {"_id": 1}},
-                    {"$project": {"name": True}},
+                    {"$match": {"$expr": {"$eq": ["$_id", "$$group_id"]}}},
+                    {"$project": {"_id": False, "id": "$_id", "name": True}},
                 ],
                 "as": set_as,
-            }
+            },
         },
+        {"$set": {set_as: {"$first": f"${set_as}"}}},
     ]
 
 
-def lookup_minimal_group_by_id(
-    local_field: str = "group", set_as: str = "group"
-) -> list[dict]:
+def lookup_groups_minimal_by_id(
+    local_field: str = "groups",
+    set_as: str = "groups",
```

Comment:
Is there somewhere where `local_field` and `set_as` arent the default values here?

Reply:
No, not that I know of, could technically be changed. Written as is to keep consistency between lookup functions